File: DGenesisWaveboxPrototypeMusicPlayer/go-backend/internal/model/Song.go
package model

import "fmt"

type Song struct {
	ID				int64
	Title			string
	Artist			string
	Album			*string
	DurationSeconds *int
	FileName		string
	OriginalFileName string
	FileSizeBytes		int64
}

func (s Song) GetFormattedDuration() string {
	if s.DurationSeconds == nil || *s.DurationSeconds <= 0 {
		return "0:00"
	}

	total := *s.DurationSeconds
	minutes := total / 60
	seconds := total % 60

	return fmt.Sprintf("%d:%02d", minutes, seconds)

}

func (s Song) IsValid() bool {
	return s.Title != "" && s.Artist != "" && s.FileName != "" 	
}

func (s Song) String() string {
	album := "None"
	if s.Album != nil {
		album = *s.Album
	}
	return fmt.Sprintf(
		"Song{id=%d, title='%s', artist='%s', album='%s', duration=%s, fileName='%s', originalFileName='%s', fileSizeBytes=%d}",
		s.ID, s.Title, s.Artist, album, s.GetFormattedDuration(), s.FileName, s.OriginalFileName, s.FileSizeBytes,
	)
}	
	

File: DGenesisWaveboxPrototypeMusicPlayer/java-backend/src/main/java/com/dgenesis/wavebox/model/Song.java
package com.dgenesis.wavebox.model;

public class Song {
    private Long id;
    private String title;
    private String artist;
    private String album;
    private String genre;
    private int durationSeconds; // Duration in seconds
    private String fileName;
    private String OriginalFileName;
    private String fileSizeBytes;

    public Song() {
    }

    public Song(Long id, String title, String artist, String album, String genre, int durationSeconds, String fileName, String originalFileName, String fileSizeBytes) {
        this.id = id;
        this.title = title;
        this.artist = artist;
        this.album = album;
        this.genre = genre;
        this.durationSeconds = durationSeconds;
        this.fileName = fileName;
        OriginalFileName = originalFileName;
        this.fileSizeBytes = fileSizeBytes;
    }

    public Long getId() {
        return id;
    }   

    public void setId(Long id) {
        this.id = id;
    }   

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public String getAlbum() {
        return album;
    }

    public void setAlbum(String album) {
        this.album = album;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public int getDurationSeconds() {
        return durationSeconds;
    }

    public void setDurationSeconds(int durationSeconds) {
        this.durationSeconds = durationSeconds;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getOriginalFileName() {
        return OriginalFileName;
    }

    public void setOriginalFileName(String originalFileName) {
        OriginalFileName = originalFileName;
    }

    public String getFileSizeBytes() {
        return fileSizeBytes;
    }

    public void setFileSizeBytes(String fileSizeBytes) {
        this.fileSizeBytes = fileSizeBytes;
    }

    
}
